### PR TITLE
fix(): added finalizer to slicegw (handled cluster detach scenario)

### PR DESCRIPTION
**Pull request by YachikaRalhan**
added finalizer to slicegw for handling cluster detach scenario

### CODE REVIEW

File: pkg/hub/controllers/slicegateway_controller.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ type SliceGwReconciler struct {
 }
 
 var sliceGWController = "workersliceGWController"
+var sliceGwFinalizer = "controller.kubeslice.io/hubSpokeSliceGw-finalizer"
```
Comment (rahulsawra98) [inline pkg/hub/controllers/slicegateway_controller.go:53]: worker.kubeslice.io/sliceGw-finalizer

File: pkg/hub/controllers/slicegateway_controller.go [hunk 2]
```diff
@@ -73,8 +74,12 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		log.Info("sliceGw doesn't belong to this cluster", "sliceGw", sliceGw.Name, "cluster", clusterName, "slicegw cluster", sliceGw.Spec.LocalGatewayConfig.ClusterName)
 		return reconcile.Result{}, nil
 	}
+	requeue, result, err := r.handleSliceGWDeletion(sliceGw, ctx, req)
```
Comment (rahulsawra98) [inline pkg/hub/controllers/slicegateway_controller.go:77]: where are you adding this finalizer?
if you are not adding how will you get an event? 
PTAL whenever you get time

**Review by rahulsawra98 — CHANGES_REQUESTED** [review]

File: pkg/hub/controllers/slicegateway_controller.go (hunk 2 above)
Comment (YachikaRalhan) [reply pkg/hub/controllers/slicegateway_controller.go:77]: Adding it inside this function on line number 213

File: pkg/hub/controllers/slicegateway_controller.go (hunk 1 above)
Comment (YachikaRalhan) [reply pkg/hub/controllers/slicegateway_controller.go:53]: I am following the same naming convention like slice controller at hub side has.
I can remove the hubSpoke after slash but the prefix should be controller IMO to make it consistent

**Review by rajendra-avesha — APPROVED** [review]

**Pull request merged**